src/storage/mod.rs

```diff
@@ -1134,14 +1134,28 @@ impl<E: Engine> Storage<E> {
         limit: usize,
         statistics: &mut Statistics,
         key_only: bool,
+        reverse: bool,
     ) -> Result<Vec<Result<KvPair>>> {
         let mut option = IterOption::default();
         if let Some(end) = end_key {
-            option.set_upper_bound(end.into_encoded());
+            if reverse {
+                option.set_lower_bound(end.into_encoded());
+            } else {
+                option.set_upper_bound(end.into_encoded());
+            }
         }
-        let mut cursor = snapshot.iter_cf(Self::rawkv_cf(cf)?, option, ScanMode::Forward)?;
+        let scan_mode = if reverse {
+            ScanMode::Backward
+        } else {
+            ScanMode::Forward
+        };
+        let mut cursor = snapshot.iter_cf(Self::rawkv_cf(cf)?, option, scan_mode)?;
         let statistics = statistics.mut_cf_statistics(cf);
-        if !cursor.seek(start_key, statistics)? {
+        if reverse {
+            if !cursor.seek_for_prev(start_key, statistics)? {
```

Comment:
Our ranges are always left-close right-open and `seek_for_prev` is inclusive. Please consider using `reverse_seek` instead.

Reply:
Also notice that your tests also need to be updated.

Reply:
fixed